File: java/library/src/main/java/com/v2soft/styxlib/library/messages/StyxTAuthMessage.java
package com.v2soft.styxlib.library.messages;

import com.v2soft.styxlib.library.io.IStyxDataReader;
import com.v2soft.styxlib.library.io.IStyxDataWriter;
import com.v2soft.styxlib.library.messages.base.StyxMessage;
import com.v2soft.styxlib.library.messages.base.StyxTMessageFID;
import com.v2soft.styxlib.library.messages.base.enums.MessageType;

import java.io.IOException;
import java.io.UnsupportedEncodingException;

public class StyxTAuthMessage extends StyxTMessageFID {
	private String mUserName;
	private String mMountPoint;
	
	public StyxTAuthMessage(long fid) {
		super(MessageType.Tauth, MessageType.Rauth, fid);
	}

    @Override
    public void load(IStyxDataReader input) 
        throws IOException  {
        super.load(input);
        setUserName(input.readUTFString());
        setMountPoint(input.readUTFString());
    }
	
	public String getUserName() {
		if (mUserName == null)
			return "";
		return mUserName;
	}
	
	public void setUserName(String user_name) {
		mUserName = user_name;
	}
	
	public String getMountPoint() {
		if (mMountPoint == null)
			return "";
		return mMountPoint;
	}
	
	public void setMountPoint(String mount_point)
	{
		mMountPoint = mount_point;
	}
	
	@Override
	public int getBinarySize() {
		return super.getBinarySize()
			+ StyxMessage.getUTFSize(getUserName())
			+ StyxMessage.getUTFSize(getMountPoint());
	}
	
	@Override
	public void writeToBuffer(IStyxDataWriter output)
	        throws UnsupportedEncodingException, IOException {
	    super.writeToBuffer(output);
		output.writeUTFString(getUserName());
		output.writeUTFString(getMountPoint());		
	}

	@Override
    public String toString() {
		return String.format("%s\nUserName: %s\nMountPoint: %s",
				super.toString(), getUserName(), getMountPoint());
	}
}

File: native/V2StyxLib/src/messages/StyxTAuthMessage.cpp
/*
 * StyxTAuthMessage.cpp
 *
 *  Created on: Apr 7, 2018
 *      Author: mrco
 */

#include "messages/StyxTAuthMessage.h"
#include <sstream>

StyxTAuthMessage::StyxTAuthMessage(StyxFID fid) :
	StyxTMessageFID(Tauth, Rauth, fid),
    mUserName(""),
	mMountPoint("") {
}

StyxTAuthMessage::~StyxTAuthMessage() {

}

StyxString StyxTAuthMessage::getMountPoint() {
	return mMountPoint;
}

StyxString StyxTAuthMessage::getUserName() {
	return mUserName;
}

void StyxTAuthMessage::setUserName(StyxString userName) {
	mUserName = userName;
}

void StyxTAuthMessage::setMountPoint(StyxString mountPoint) {
	mMountPoint = mountPoint;
}

void StyxTAuthMessage::load(IStyxDataReader *buffer) {
	StyxTMessageFID::load(buffer);
	setUserName(buffer->readUTFString());
	setMountPoint(buffer->readUTFString());
}

void StyxTAuthMessage::writeToBuffer(IStyxDataWriter* output) {
	StyxTMessageFID::writeToBuffer(output);
	output->writeUTFString(getUserName());
	output->writeUTFString(getMountPoint());
}

size_t StyxTAuthMessage::getBinarySize() {
	return StyxTMessageFID::getBinarySize()
		+ StyxMessage::getUTFSize(getUserName())
		+ StyxMessage::getUTFSize(getMountPoint());
}

StyxString StyxTAuthMessage::toString() {
	std::stringstream stream;
	stream << StyxTMessageFID::toString() << "\nUserName: "
			<< getUserName() << "\nMountPoint: " << getMountPoint();
	return stream.str();
}
